notify-deploy-to-slack/action.yml

```
@@ -0,0 +1,49 @@
+# Copyright (c) 2024 Hemi Labs, Inc.
+# Use of this source code is governed by the MIT License,
+# which can be found in the LICENSE file.
+
+name: Notify to Slack
+
+author: Gabriel Montes
+
+description: Send a notification to Slack using incoming webhooks
+
+inputs:
+  app-name:
+    description: "The app being deployed"
+    required: true
+  environment:
+    description: "The deployment environment"
+    required: true
+  reference:
+    description: "Additional reference to add to the notification"
+    required: false
+    default: ""
+  slack-mention:
+    description: "The mention to include in the message"
+    default: <!here>
+  slack-webhook-url:
+    description: "The Slack incoming webhook URL"
+    required: true
+  status:
+    description: "The deployment status"
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: slackapi/slack-github-action@v1
```

As this is a third-party action, not maintained by either GitHub or Docker, we should pin it to a commit hash  in order to make it immutable.

Also, the latest version of this action is v2.1.0 (https://github.com/slackapi/slack-github-action/releases/tag/v2.1.0) - are we able to update? There is a migration guide here: https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0

```suggestion
      - uses: slackapi/slack-github-action@fcfb566f8b0aab22203f066d80ca1d7e4b5d05b3 # v1.27.1
```

https://github.com/slackapi/slack-github-action/releases/tag/v1.27.1
https://github.com/slackapi/slack-github-action/commit/fcfb566f8b0aab22203f066d80ca1d7e4b5d05b3